packages/hardhat-core/src/internal/core/jsonrpc/types/output/decodeJsonRpcResponse.ts

```diff
@@ -0,0 +1,15 @@
+import * as t from "io-ts";
+
+import { InvalidResponseError } from "../../../providers/errors";
+
+/**
+ * This function decodes an RPC out type, throwing InvalidResponseError if it's not valid.
+ */
+export function decodeJsonRpcResponse<T>(value: unknown, codec: t.Type<T>) {
+  return codec.decode(value).fold(() => {
+    // tslint:disable-next-line
+    throw new InvalidResponseError(
```

Comment:
I invented this type of error. I think it makes sense to throw a `ProviderError` if the provider received an incorrect response.